File: YT3Finder/app/src/main/java/com/yt3/finder/MainActivity.java
package com.yt3.finder;

import android.Manifest;
import android.app.*;
import android.content.*;
import android.content.pm.PackageManager;
import android.graphics.Color;
import android.net.Uri;
import android.os.*;
import android.widget.*;

public class MainActivity extends Activity {
    private SharedPreferences prefs;
    private TextView stats,status,candidateText;
    private Button start,pause,copy,verify,reset;
    private final Handler handler=new Handler(Looper.getMainLooper());

    private int dp(int n){return (int)(n*getResources().getDisplayMetrics().density+.5f);}
    private TextView text(String s,int size,int color){TextView v=new TextView(this);v.setText(s);v.setTextSize(size);v.setTextColor(color);v.setPadding(dp(4),dp(6),dp(4),dp(6));return v;}
    private Button btn(String s){Button b=new Button(this);b.setText(s);b.setAllCaps(false);b.setTextSize(15);return b;}
    private LinearLayout card(){LinearLayout l=new LinearLayout(this);l.setOrientation(LinearLayout.VERTICAL);l.setPadding(dp(16),dp(14),dp(16),dp(14));l.setBackgroundColor(Color.rgb(18,23,33));LinearLayout.LayoutParams p=new LinearLayout.LayoutParams(-1,-2);p.setMargins(dp(12),dp(8),dp(12),dp(8));l.setLayoutParams(p);return l;}

    @Override public void onCreate(Bundle b){
        super.onCreate(b);
        getWindow().setStatusBarColor(Color.rgb(8,10,15));getWindow().setNavigationBarColor(Color.rgb(8,10,15));
        prefs=getSharedPreferences("yt3",MODE_PRIVATE);
        if(Build.VERSION.SDK_INT>=33&&checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS)!=PackageManager.PERMISSION_GRANTED)
            requestPermissions(new String[]{Manifest.permission.POST_NOTIFICATIONS},5);

        LinearLayout root=new LinearLayout(this);root.setOrientation(LinearLayout.VERTICAL);root.setBackgroundColor(Color.rgb(8,10,15));
        ScrollView sc=new ScrollView(this);sc.addView(root);setContentView(sc);

        TextView title=text("YT3 Finder",30,Color.WHITE);title.setTypeface(null,1);title.setPadding(dp(16),dp(20),dp(16),dp(4));root.addView(title);
        root.addView(text("Цель: найти хоть один доступный латинский трёхзнак",14,Color.rgb(145,154,174)));

        LinearLayout stat=card();
        stat.addView(text("Поиск по всем 51 840 вариантам",20,Color.WHITE));
        stats=text("0 / 51 840",28,Color.WHITE);stats.setTypeface(null,1);stat.addView(stats);
        status=text("Готово к поиску.",14,Color.rgb(145,154,174));stat.addView(status);
        LinearLayout actions=new LinearLayout(this);actions.setOrientation(LinearLayout.HORIZONTAL);
        start=btn("▶ Начать");pause=btn("Пауза");reset=btn("Сброс");
        actions.addView(start,new LinearLayout.LayoutParams(0,-2,1));actions.addView(pause,new LinearLayout.LayoutParams(0,-2,1));actions.addView(reset,new LinearLayout.LayoutParams(0,-2,1));
        stat.addView(actions);root.addView(stat);

        LinearLayout found=card();
        found.addView(text("Найденный кандидат",20,Color.WHITE));
        candidateText=text("—",36,Color.rgb(244,200,74));candidateText.setTypeface(null,1);candidateText.setGravity(17);candidateText.setPadding(dp(8),dp(18),dp(8),dp(18));found.addView(candidateText);
        copy=btn("Копировать");verify=btn("Проверить / поставить в YouTube");
        found.addView(copy);found.addView(verify);root.addView(found);

        LinearLayout info=card();
        info.addView(text("Приложение больше не ищет красивые варианты и не тратит время на другие языки. Оно перебирает весь латинский набор в стабильном случайном порядке, чтобы не начинать с очевидно занятых комбинаций. Публичные проверки используются только как жёсткий предварительный отсев. Финальную доступность может подтвердить только сам YouTube при попытке сохранить handle. Если YouTube не принимает кандидат — нажми «Искать следующий», и поиск продолжится с места остановки.",13,Color.rgb(215,220,230)));
        root.addView(info);

        start.setOnClickListener(v->startScan());
        pause.setOnClickListener(v->sendAction(ScanService.ACTION_PAUSE));
        reset.setOnClickListener(v->{sendAction(ScanService.ACTION_RESET);handler.postDelayed(this::refresh,350);});
        copy.setOnClickListener(v->copyCandidate());
        verify.setOnClickListener(v->openYouTubeHandle());

        handler.post(refreshLoop);
    }

    private void startScan(){
        Intent i=new Intent(this,ScanService.class);i.setAction(ScanService.ACTION_START);
        if(Build.VERSION.SDK_INT>=26)startForegroundService(i);else startService(i);
        handler.postDelayed(this::refresh,300);
    }
    private void sendAction(String action){Intent i=new Intent(this,ScanService.class);i.setAction(action);startService(i);}
    private final Runnable refreshLoop=new Runnable(){public void run(){refresh();handler.postDelayed(this,800);}};

    private void refresh(){
        int index=prefs.getInt("index",0),total=prefs.getInt("total",51840);
        boolean running=prefs.getBoolean("running",false);
        String c=prefs.getString("candidate","");
        stats.setText(index+" / "+total);
        candidateText.setText(c.isEmpty()?"—":"@"+c);
        copy.setEnabled(!c.isEmpty());verify.setEnabled(!c.isEmpty());pause.setEnabled(running);
        start.setEnabled(!running);
        start.setText(c.isEmpty()?"▶ Начать / продолжить":"Искать следующий");
        if(running)status.setText("Ищу в фоне… можно свернуть приложение.");
        else if(!c.isEmpty())status.setText("Кандидат найден. Проверь его непосредственно в YouTube.");
        else status.setText(prefs.getString("error",index>0?"Пауза. Можно продолжить.":"Готово к поиску."));
    }

    private void copyCandidate(){
        String c=prefs.getString("candidate","");if(c.isEmpty())return;
        ((android.content.ClipboardManager)getSystemService(CLIPBOARD_SERVICE)).setPrimaryClip(ClipData.newPlainText("handle","@"+c));
        Toast.makeText(this,"Скопировано @"+c,Toast.LENGTH_SHORT).show();
    }
    private void openYouTubeHandle(){
        String c=prefs.getString("candidate","");if(c.isEmpty())return;
        copyCandidate();
        try{startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.youtube.com/handle")));}
        catch(Exception e){startActivity(new Intent(Intent.ACTION_VIEW,Uri.parse("https://www.youtube.com/@"+Uri.encode(c))));}
    }
    @Override protected void onDestroy(){super.onDestroy();handler.removeCallbacksAndMessages(null);}
}

File: tgfinder/app/src/main/java/com/tgfinder/app/MainActivity.kt
package com.tgfinder.app

import android.Manifest
import android.content.*
import android.os.Build
import android.os.Bundle
import android.text.InputType
import android.view.Gravity
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import java.text.SimpleDateFormat
import java.util.*
import kotlin.concurrent.thread

class MainActivity:AppCompatActivity(){
    private lateinit var root:LinearLayout;private lateinit var status:TextView;private lateinit var foundBox:LinearLayout
    private fun dp(v:Int)=(v*resources.displayMetrics.density).toInt()
    override fun onCreate(b:Bundle?){super.onCreate(b);if(Build.VERSION.SDK_INT>=33)ActivityCompat.requestPermissions(this,arrayOf(Manifest.permission.POST_NOTIFICATIONS),1);buildUi();TelegramCore.listen{runOnUiThread{refreshState()}}; val id=Prefs.apiId(this); if(id>0)TelegramCore.start(this,id,Prefs.apiHash(this));refreshFound()}
    private fun buildUi(){
        val scroll=ScrollView(this);root=LinearLayout(this).apply{orientation=LinearLayout.VERTICAL;setPadding(dp(14),dp(18),dp(14),dp(40));setBackgroundColor(0xffeef7fc.toInt())};scroll.addView(root);setContentView(scroll)
        title("Telegram Username Finder",24); text("Зелёный результат = Telegram подтвердил доступность через TDLib.")
        val api=edit("API ID",InputType.TYPE_CLASS_NUMBER);api.setText(Prefs.apiId(this).takeIf{it>0}?.toString()?:"")
        val hash=edit("API HASH");hash.setText(Prefs.apiHash(this));val phone=edit("Телефон +...",InputType.TYPE_CLASS_PHONE);phone.setText(Prefs.phone(this))
        button("Подключить Telegram"){val id=api.text.toString().toIntOrNull()?:0;val h=hash.text.toString().trim();val p=phone.text.toString().trim();Prefs.saveCredentials(this,id,h,p);TelegramCore.start(this,id,h);Toast.makeText(this,"Инициализация…",Toast.LENGTH_SHORT).show()}
        status=TextView(this).apply{setPadding(0,dp(8),0,dp(8));textSize=15f};root.addView(status)
        button("Отправить номер"){TelegramCore.submitPhone(phone.text.toString().trim()){e->runOnUiThread{toast(e?:"Код отправлен")}}}
        val code=edit("Код из Telegram",InputType.TYPE_CLASS_NUMBER);button("Подтвердить код"){TelegramCore.submitCode(code.text.toString().trim()){e->runOnUiThread{toast(e?:"Код принят")}}}
        val pass=edit("Пароль 2FA",InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD);button("Подтвердить 2FA"){TelegramCore.submitPassword(pass.text.toString()){e->runOnUiThread{toast(e?:"Готово")}}}
        title("Ручная точная проверка",19);val manual=edit("apple / aaaaaa");button("Проверить"){thread{val u=manual.text.toString().trim().removePrefix("@").lowercase();val r=TelegramCore.checkUsername(u);runOnUiThread{toast("@$u — ${r.label}");if(r.status=="free"){val l=Prefs.loadFound(this);if(l.none{it.username==u}){l.add(0,Prefs.Found(u,System.currentTimeMillis()));Prefs.saveFound(this,l);refreshFound()}}}}}
        title("Автопоиск",19)
        val modes=arrayOf("Английские слова","Красивые ID","Всё вместе");val spin=Spinner(this);spin.adapter=ArrayAdapter(this,android.R.layout.simple_spinner_dropdown_item,modes);spin.setSelection(when(Prefs.mode(this)){"pretty"->1;"mixed"->2;else->0});root.addView(spin)
        val delays=arrayOf("Быстро · 0.6с","Нормально · 0.9с","Осторожно · 1.3с");val ds=Spinner(this);ds.adapter=ArrayAdapter(this,android.R.layout.simple_spinner_dropdown_item,delays);root.addView(ds)
        button("▶ Начать фоновый поиск"){Prefs.setMode(this,when(spin.selectedItemPosition){1->"pretty";2->"mixed";else->"words"});Prefs.setDelay(this,longArrayOf(600,900,1300)[ds.selectedItemPosition]);val i=Intent(this,ScanService::class.java);if(Build.VERSION.SDK_INT>=26)startForegroundService(i)else startService(i);toast("Поиск работает в фоне")}
        button("■ Остановить"){startService(Intent(this,ScanService::class.java).setAction("STOP"))}
        title("Подтверждённо свободные",19);foundBox=LinearLayout(this).apply{orientation=LinearLayout.VERTICAL};root.addView(foundBox)
    }
    private fun refreshState(){status.text=TelegramCore.statusText}
    private fun refreshFound(){if(!::foundBox.isInitialized)return;foundBox.removeAllViews();val list=Prefs.loadFound(this);if(list.isEmpty()){textInto(foundBox,"Пока ничего не найдено.");return};val fmt=SimpleDateFormat("dd.MM HH:mm",Locale.getDefault());list.take(200).forEach{f->
        val row=LinearLayout(this).apply{orientation=LinearLayout.HORIZONTAL;gravity=Gravity.CENTER_VERTICAL;setPadding(dp(10),dp(9),dp(6),dp(9));setBackgroundColor(0xffe3f8ef.toInt())}
        val tv=TextView(this).apply{text="@${f.username}\n${fmt.format(Date(f.checkedAt))}";textSize=16f;setTextColor(0xff08764b.toInt())};row.addView(tv,LinearLayout.LayoutParams(0,ViewGroup.LayoutParams.WRAP_CONTENT,1f))
        val copy=Button(this).apply{text="Копировать";setOnClickListener{(getSystemService(CLIPBOARD_SERVICE) as android.content.ClipboardManager).setPrimaryClip(android.content.ClipData.newPlainText("username","@${f.username}"));toast("Скопировано")}};row.addView(copy);foundBox.addView(row,LinearLayout.LayoutParams(-1,-2).apply{bottomMargin=dp(7)})
    }}
    private fun title(s:String,size:Int){root.addView(TextView(this).apply{text=s;textSize=size.toFloat();setTextColor(0xff10233f.toInt());setPadding(0,dp(12),0,dp(7));setTypeface(typeface,1)})}
    private fun text(s:String){textInto(root,s)}
    private fun textInto(v:LinearLayout,s:String){v.addView(TextView(this).apply{text=s;textSize=14f;setTextColor(0xff70819b.toInt());setPadding(0,0,0,dp(8))})}
    private fun edit(h:String,type:Int=InputType.TYPE_CLASS_TEXT)=EditText(this).apply{hint=h;inputType=type;setPadding(dp(12),dp(10),dp(12),dp(10));root.addView(this,LinearLayout.LayoutParams(-1,-2).apply{bottomMargin=dp(8)})}
    private fun button(t:String,on:()->Unit){root.addView(Button(this).apply{text=t;setOnClickListener{on()}},LinearLayout.LayoutParams(-1,-2).apply{bottomMargin=dp(7)})}
    private fun toast(s:String)=Toast.makeText(this,s,Toast.LENGTH_LONG).show()
    override fun onResume(){super.onResume();refreshFound();refreshState()}
}
